### PR TITLE
feat(api): G-Code CLI

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Add G-Code CLI and documentation for using it

# Changelog

* Create CLI
* Add tests for CLI
* Add documentation for CLI
* Add dash separator to Default text mode and update tests
* Change smoothie_config param in ProtocolRunner to Settings type. Previous SmoothieSettings type was incorrect
* Add save_text_explanation_to_file method to GCodeProgram
* Update save_json_to_file method to take a file path instead of file name in GCodeProgram

# Review requests

None

# Risk assessment

None

### CODE REVIEW

File: api/src/opentrons/hardware_control/g_code_parsing/cli.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+             '$DATA_DIR/g_code_validation_protocols/smoothie_protocol.py'
+        ]
+        """
+        parsed_dict = vars(cls.parser().parse_args(args))
+        return cls._post_process_args(parsed_dict)
+
+    @classmethod
+    def to_command(
+            cls, processed_args
+    ) -> Union[RunCommand, DiffCommand]:
```
Comment (amitlissack) [inline api/src/opentrons/hardware_control/g_code_parsing/cli.py]: Could this return `CLICommand` instead?

File: api/src/opentrons/hardware_control/g_code_parsing/cli.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+    @classmethod
+    def parser(cls) -> argparse.ArgumentParser:
+        """
+        Method to generate argparse ArgumentParser class for parsing command line
+        input
+        :return: Parser object
+        """
+        parser = argparse.ArgumentParser(
+            description='CLI for interfacing G-Code Parser with S3'
```
Comment (amitlissack) [inline api/src/opentrons/hardware_control/g_code_parsing/cli.py]: Sorry, but what is `S3` meaning here?

Comment (DerekMaggio) [reply api/src/opentrons/hardware_control/g_code_parsing/cli.py]: So we will be storing the generated G-Code description to S3 for comparison between versions.

Comment (DerekMaggio) [reply api/src/opentrons/hardware_control/g_code_parsing/cli.py]: But I guess it is not interfacing with S3 at all so I will remove that

**Review by amitlissack — APPROVED** [review]
Rockin!

**Pull request merged**